__tests__/controllers/auth/create-account.spec.ts

```
@@ -1,30 +1,79 @@
 import request from 'supertest';
 import app from '@app';
-import { factory } from 'typeorm-seeding';
-import { User } from '@entities/user.entity';
 import { API } from '../../utils';
 import { ErrorsMessages } from '@constants/errorMessages';
 import { HttpStatusCode } from '@constants/httpStatusCode';
+import { AuthController } from '../../../src/controllers/auth.controller';
+import { Container } from 'typedi';
+import { SessionService } from '@services/session.service';
+import { User } from '@entities/user.entity';
+import { factory } from 'typeorm-seeding';
+import { SignUpDTO } from '@dto/signUpDTO';
+import { mockResponse, mockRequest } from '../../utils/mocks';
+import { Request, Response } from 'express';
+import { EmailService } from '@services/email.service';
+
+let authController: AuthController;
+let sessionService: SessionService;
+let userFields;
 
 describe('creating an account', () => {
+  beforeAll(async () => {
+    authController = Container.get(AuthController);
+    sessionService = Container.get(SessionService);
+    userFields = await factory(User)().make();
+  });
+
+  it('all dependencies should be defined', () => {
+    expect(authController).toBeDefined();
+    expect(sessionService).toBeDefined();
+  });
+
+  it('sends the verification email', async () => {
+    const signUPDTO = await factory(SignUpDTO)().make();
+
+    jest.spyOn(sessionService, 'signUp').mockResolvedValueOnce(userFields);
+    jest.spyOn(EmailService, 'sendEmail').mockResolvedValueOnce(true);
+
+    const response = await authController.signUp(
+      signUPDTO,
+      mockRequest as Request,
+      mockResponse as Response
+    );
+    expect(response).toHaveProperty('email');
+  });
+
+  it('does not send the verification email', async () => {
```

A better name for this test case could be "it('should throw an error if something fails while sending the email')" what do you think?

The main goal is to put descriptive names to test cases, describing what is the situation that someone should expect when reading the case.